### PR TITLE
optional graph parameter to create the database

### PR DESCRIPTION
**Pull request by baaa**
optional graph parameter to create the database, also refactored mutable futures from graph init

### CODE REVIEW

File: driver/src/test/scala/spec/GraphSpec.scala [hunk 1]
```diff
@@ -18,6 +19,15 @@ class GraphSpec extends AsyncWordSpec with Matchers {
         s.token shouldNot be("")
       }
     }
+//    "create the database" ignore  {
```
Comment (darkfrog26) [inline driver/src/test/scala/spec/GraphSpec.scala]: if you're going to add support for creating a database, you should also add support for dropping one or this is an incomplete PR

**Comment by darkfrog26** [comment]
Any updates on this or did you decide to abandon the effort?

**Comment by baaa** [comment]
its sort of on pause at the mo. I completely abandoned scarango and using rest api since I need to push live soon. but planning to come back and change many things

**Comment by darkfrog26** [comment]
I'm sorry to hear that you've had to go directly to REST.  When you're ready to come back to Scarango let me know and I'd be happy to help work through the problems with you.

**Comment by darkfrog26** [comment]
Closing this PR as I've just added complete database management functionality to the driver.

**Pull request closed without merging**